modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medguns.dm

```diff
@@ -0,0 +1,213 @@
+//Base Medigun Code//
+/obj/item/gun/energy/medigun
+	name = "MediGun"
+	desc = "This is my smart gun, it won't hurt anyone friendly, infact it will make them heal! Please tell github if you somehow manage to get this gun."
+	icon = 'modular_skyrat/modules/modular_weapons/icons/obj/guns/projectile.dmi'
+	icon_state = "cfa-disabler"
+	inhand_icon_state = null
+	ammo_type = list(/obj/item/ammo_casing/energy/medical) //The default option that heals Oxygen//
+	ammo_x_offset = 2
+	w_class = WEIGHT_CLASS_NORMAL
+	cell_type = /obj/item/stock_parts/cell/medigun/
+	ammo_x_offset = 2
+	charge_sections = 3
+	has_gun_safety = TRUE
+	var/maxcells = 3
+	var/cellcount = 0
+	var/list/installedcells = list()
```

Comment:
You can get rid of installedcells and use the guns contents list , instead of making a new one , and for the cell count you could could just check its length

Reply:
no pressure to do it , it saves very little performance however.

Reply:
Aye, I might look into that in a bit. Thank you for the suggestion!